flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/ProcessMemoryUtils.java

```diff
@@ -0,0 +1,281 @@
+		MemorySize minSize = getMemorySizeFromConfig(config, options.getJvmOptions().getJvmOverheadMin());
+		MemorySize maxSize = getMemorySizeFromConfig(config, options.getJvmOptions().getJvmOverheadMax());
+		return getRangeFraction(minSize, maxSize, options.getJvmOptions().getJvmOverheadFraction(), config);
+	}
+
+	public static MemorySize getMemorySizeFromConfig(Configuration config, ConfigOption<MemorySize> option) {
+		try {
+			return Preconditions.checkNotNull(config.get(option), "The memory option is not set and has no default value.");
+		} catch (Throwable t) {
+			throw new IllegalConfigurationException("Cannot read memory size from config option '" + option.key() + "'.", t);
+		}
+	}
+
+	public static RangeFraction getRangeFraction(
+			MemorySize minSize,
+			MemorySize maxSize,
+			ConfigOption<Float> fractionOption,
+			Configuration config) {
+		double fraction = config.getFloat(fractionOption);
+		try {
+			return new RangeFraction(minSize, maxSize, fraction);
+		} catch (IllegalArgumentException e) {
+			throw new IllegalConfigurationException(
+				String.format(
+					"Inconsistently configured %s (%s) and its min (%s), max (%s) value",
+					fractionOption,
+					fraction,
+					minSize.toHumanReadableString(),
+					maxSize.toHumanReadableString()),
+				e);
+		}
+	}
+
+	public static MemorySize deriveWithFraction(String memoryDescription, MemorySize base, RangeFraction rangeFraction) {
+		MemorySize relative = base.multiply(rangeFraction.getFraction());
+		return capToMinMax(memoryDescription, relative, rangeFraction);
+	}
+
+	public static MemorySize deriveWithInverseFraction(String memoryDescription, MemorySize base, RangeFraction rangeFraction) {
+		checkArgument(rangeFraction.getFraction() < 1);
+		MemorySize relative = base.multiply(rangeFraction.getFraction() / (1 - rangeFraction.getFraction()));
+		return capToMinMax(memoryDescription, relative, rangeFraction);
+	}
+
+	private static MemorySize capToMinMax(
+			String memoryDescription,
+			MemorySize relative,
+			RangeFraction rangeFraction) {
+		long size = relative.getBytes();
+		if (size > rangeFraction.getMaxSize().getBytes()) {
+			LOG.info(
+				"The derived from fraction {} ({}) is greater than its max value {}, max value will be used instead",
+				memoryDescription,
+				relative.toHumanReadableString(),
+				rangeFraction.getMaxSize().toHumanReadableString());
+			size = rangeFraction.getMaxSize().getBytes();
+		} else if (size < rangeFraction.getMinSize().getBytes()) {
+			LOG.info(
+				"The derived from fraction {} ({}) is less than its min value {}, min value will be used instead",
+				memoryDescription,
+				relative.toHumanReadableString(),
+				rangeFraction.getMinSize().toHumanReadableString());
+			size = rangeFraction.getMinSize().getBytes();
+		}
+		return new MemorySize(size);
+	}
+
+	public static String generateJvmParametersStr(ProcessMemorySpec processSpec) {
+		return "-Xmx" + processSpec.getJvmHeapMemorySize().getBytes()
```

Comment:
@azagrebin @xintongsong 

Hi, I'm upgrading our Flink to 1.11 version. And I find that the memory related arguments in JVM parameters are hard to read here because we're using bytes here, for example:

```
jdk/jdk1.8/bin/java -Xmx944892739 -Xms944892739 -XX:MaxDirectMemorySize=1213328302
```

I wonder whether we can make some minor changes here to improve this. Or did you use bytes on purpose for other reasons I probably missed?

Reply:
@Jiayi-Liao,

I believe these parameters are intended to use bytes, in order to preserve the accuracy. We have run into troubles previously caused by rounding errors when using megabytes. So we decided to always use bytes for Flink's internal memory setups and calculations.

I agree with you that it would be helpful to have some human readable representations for the memory sizes and JVM parameters. But I think the parameters here are intended to be read by the JVM rather than users. How about we print such information in the beginning of the logs?

@azagrebin 

I think we probably should also log the parameters/configurations in a human readable way in `ProcessMemoryUtils.generateJvmParametersStr` and `TaskExecutorProcessUtils.generateDynamicConfigsStr`. WDYT?

Reply:
@xintongsong @Jiayi-Liao 
I agree we can log the human readable values for JVM memory arguments as they are not always just option values.
We can also do it for  `generateDynamicConfigsStr` on debug level, the dynamic properties will be also outputted in startup logs of the TM JVM process itself.

Reply:
FYI, I've created FLINK-18175 to track this.